### DISCUSSION TITLE
Maui much slower then Forms in debug-mode

### DISCUSSION THREAD

**Discussion by AlexanderVanMeerten** [dotnet/maui#15722] [Upgrading from Xamarin]
### Description

I have been porting my Forms app to Maui. It's is not finished yet but debugging Maui is so much slower then Forms.
I made a screen recording. On the left is Forms, on the right is Maui. 

Macbook Pro i9. For Android it is the same.

### Steps to Reproduce

See recording: https://github.com/dotnet/maui/assets/10942447/2a90878f-e20f-4ad7-9b0d-140e2dc8d186

### Link to public reproduction project repository

https://github.com/dotnet/maui/assets/10942447/2a90878f-e20f-4ad7-9b0d-140e2dc8d186

### Version with bug

7.0.86

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

iOS simulator, Android emmulator

**Comment by drasticactions** [comment]
I converted this to a discussion, as there aren't actionable items here to "fix" yet, especially if several issues are affecting your performance. If you could provide specific reproduction steps showing your performance issues and making individual issues for those, and use this discussion to talk about the general issues you're having with perf, then that would help fix what you're seeing.

**Reply by roubachof** [reply]
XAMLC Is disabled in DEBUG on maui, so the view construction is reaaaallly slow. Good news is that you can re-enable it with:
```xml
<_MauiForceXamlCForDebug>true</_MauiForceXamlCForDebug>
```

**Reply by AlexanderVanMeerten** [reply]
That sounds great! Where can I put this tag?

**Reply by roubachof** [reply]
Somthing like that I guess:

```xml
<PropertyGroup Condition=" '$(Configuration)' == 'Debug'>
    <_MauiForceXamlCForDebug>true</_MauiForceXamlCForDebug>
    ...
</PropertyGroup>
```

source: https://www.sharpnado.com/net-maui-painfully-slow-debugging/

**Reply by davidortinau** [reply]
@roubachof I just read your blog and was about to post it here!

**Reply by developer9969** [reply]
I was very excited about this findings however my hotreload does not work, removed "_MauiForceXamlCForDebug" and worked again.. but because hot reload is a bit temperamental.. may be someone else can confirm that this is the case, which would make this change not very usable if you are tweaking some pages.

**Reply by jonathanpeppers** [reply]
In addition to changing the XamlC settings, the Mono intepreter will have an impact on performance. `UseInterpreter=true` is on by default for Debug mode, as it enables the C# hot reload feature.

Information on this at the top here: http://aka.ms/profile-maui

**Reply by developer9969** [reply]
@jonathanpeppers  those kind of blogs are extremely useful and explains a lot, however fact of the matter is that the Dev loop is incredibly slow in Maui, I have a solution with 20 projects with just iOS and Android as platform and it takes on android good few mins to be in a debug mode again, so if you are tweaking a page and hot reload is a bit of a "hit and miss" it affects us big time..

Do you have any paper/blog on how to debug what is wrong when full link is enabled , we would like to reduce the size of the app that in Maui is nearly double the size of Xamarin . My iOS app is reduced by 20 mg but it crashes .. and I would like to do the same with proguard but no many blogs about how to go about it and what to exclude, debug..

thanks

**Reply by jonathanpeppers** [reply]
> Do you have any paper/blog on how to debug what is wrong when full link is enabled

MAUI is not yet "trimming safe", this is something coming in .NET 9. If you search PRs with the word "trimming", you can see some of the progress:

* https://github.com/dotnet/maui/pulls?q=is%3Apr+trimming

This is more important for us on mobile because app size is crucial, but I don't think WPF, WinUI, etc. are trimming safe either.

**Comment by developer9969** [comment]
thank you for your time in answering.. I guess we have to wait for trimming then.. WPF etc.. they dont give a monkey about the size of the app.. for us is crucial and just hope my clients dont moan when they see the size of the maui app..
